File: 8vim/src/main/java/inc/flide/vim8/structures/yaml/Layout.java
package inc.flide.vim8.structures.yaml;

import com.fasterxml.jackson.annotation.JsonProperty;

public class Layout {
    @JsonProperty(required = true)
    public Layers layers;
    public LayoutInfo info = new LayoutInfo();

    public static class LayoutInfo {
        public String name = "";
        public String description = "";
        public Contact contact = new Contact();

        public static class Contact {
            public String name = "";
            public String email = "";
        }
    }
}

File: 8vim/src/main/kotlin/inc/flide/vim8/models/yaml/Layout.kt
package inc.flide.vim8.models.yaml

import arrow.optics.optics
import com.fasterxml.jackson.annotation.JsonProperty

@optics
data class Layout(
    @JsonProperty(required = true) val layers: Layers = Layers(),
    val info: LayoutInfo = LayoutInfo()
) {
    companion object
}

@optics
data class LayoutInfo(
    val name: String = "",
    val description: String = "",
    val rtl: Boolean = false,
    val contact: Contact = Contact()
) {
    companion object
}

@optics
data class Contact(val name: String = "", val email: String = "") {
    companion object
}
